ui/src/app/job-list/table/table.component.spec.ts

```diff
@@ -279,6 +279,103 @@ describe('JobsTableComponent', () => {
     });
   }))
 
+  it('should select multiple jobs on shift-click', async( () => {
+    fixture.detectChanges();
+    let de: DebugElement = fixture.debugElement;
+    const checkboxes = de.queryAll(By.css('.mat-checkbox-input'));
+
+    checkboxes[2].nativeElement.click();
+    fixture.detectChanges();
+    const shiftClick = new MouseEvent('click', {
+      'button' : 0,
+      'buttons' : 0,
+      'bubbles': true,
+      'cancelable' : false,
+      'shiftKey': true
+    });
+    testComponent.jobsTableComponent.updateCheckBoxSelection(jobs[3], shiftClick);
+    fixture.detectChanges();
+
+    expect(testComponent.jobsTableComponent.allSelected()).toEqual(false);
+    expect(checkboxes[2].nativeElement.checked).toEqual(true);
+    expect(checkboxes[3].nativeElement.checked).toEqual(true);
+    expect(checkboxes[4].nativeElement.checked).toEqual(true);
+    expect(checkboxes[5].nativeElement.checked).toEqual(false);
+  }))
+
+  it('should select multiple jobs on shift-click (reverse order)', async( () => {
+    fixture.detectChanges();
+    let de: DebugElement = fixture.debugElement;
+    const checkboxes = de.queryAll(By.css('.mat-checkbox-input'));
+
+    checkboxes[5].nativeElement.click();
+    fixture.detectChanges();
+    const shiftClick = new MouseEvent('click', {
+      'button' : 0,
+      'buttons' : 0,
+      'bubbles': true,
+      'cancelable' : false,
+      'shiftKey': true
+    });
+    testComponent.jobsTableComponent.updateCheckBoxSelection(jobs[2], shiftClick);
+    fixture.detectChanges();
+
+    expect(checkboxes[1].nativeElement.checked).toEqual(false);
+    expect(checkboxes[2].nativeElement.checked).toEqual(false);
```

Comment:
The 1-based index is confusing here. Perhaps you could do a `checkboxes.shift()` to get rid of the "select all" checkbox so these indices match up with `jobs`. Or else modify your selector for checkbox inputs